### PR TITLE
refactor: volatile implementation for catalog

### PR DESCRIPTION
**Pull request by ShiKaiWi**
# Which issue does this PR close?

Part of #92 

# Rationale for this change
Preparations for integrating cluster module to CeresDB's startup in cluster mode.

# What changes are included in this PR?
Refactoring volitale implementations for catalog:
- Rename module name `memory` to `volatile`;
- Add initialization procedure for volatile catalog;
- Adjust the method signature of `alloc_schema_id` & `alloc_table_id`;

# Are there any user-facing changes?
None.

# How does this change test
Existing UT.

### CODE REVIEW

File: catalog_impls/src/volatile.rs [hunk 1]
```diff
@@ -1,8 +1,7 @@
 // Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
 
-//! A memory catalog implementation
-//!
-//! Mainly for test
```
Comment (waynexia) [inline catalog_impls/src/volatile.rs:5]: :rofl::rofl:

File: catalog_impls/src/volatile.rs [hunk 2]
```diff
@@ -1,8 +1,7 @@
...
+//! A volatile catalog implementation used for storing informations about table
+//! and schema at other components.
```
Comment (waynexia) [inline catalog_impls/src/volatile.rs:4]:
```suggestion
//! A volatile catalog implementation is used for storing information about table
//! and schema at memory.
```

File: catalog_impls/src/volatile.rs [hunk 3]
```diff
@@ -17,51 +16,82 @@ use catalog::{
...
 };
-use log::info;
+use log::{debug, info};
 use snafu::{ensure, ResultExt};
 use table_engine::table::{SchemaId, TableId, TableRef};
 use tokio::sync::Mutex;
 
 #[async_trait]
-pub trait SchemaIdGenerator {
+pub trait SchemaIdAlloc: Send + Sync {
```
Comment (waynexia) [inline catalog_impls/src/volatile.rs:29]: Should we move these common traits (`SchemaIdAlloc` and `TableIdAlloc`) to `catalog_impl/lib.rs`?

File: catalog_impls/src/volatile.rs [hunk 4]
```diff
@@ -17,51 +16,82 @@ use catalog::{
...
 #[async_trait]
-pub trait TableIdGenerator {
+pub trait TableIdAlloc: Send + Sync {
     type Error: std::error::Error + Send + Sync + 'static;
     async fn alloc_table_id(
         &self,
-        schema_id: SchemaId,
-        table_name: NameRef,
+        schema_name: &str,
+        table_name: &str,
```
Comment (waynexia) [inline catalog_impls/src/volatile.rs:43]:
```suggestion
        schema_name: NameRef,
        table_name: NameRef,
```

File: catalog_impls/src/volatile.rs [hunk 5]
```diff
@@ -17,51 +16,82 @@ use catalog::{
...
     ) -> std::result::Result<TableId, Self::Error>;
 
     async fn invalidate_table_id(
         &self,
-        schema_id: SchemaId,
+        schema_name: &str,
+        table_name: &str,
```
Comment (waynexia) [inline catalog_impls/src/volatile.rs:49]:
```suggestion
        schema_name: NameRef,
        table_name: NameRef,
```

File: catalog_impls/src/volatile.rs [hunk 6]
```diff
@@ -17,51 +16,82 @@ use catalog::{
...
-        schema_id: SchemaId,
-        table_name: NameRef,
+        schema_name: &str,
+        table_name: &str,
     ) -> std::result::Result<TableId, Self::Error>;
 
     async fn invalidate_table_id(
         &self,
-        schema_id: SchemaId,
+        schema_name: &str,
```
Comment (waynexia) [inline catalog_impls/src/volatile.rs:48]: Why we deprecate schema id here

Comment (ShiKaiWi) [reply catalog_impls/src/volatile.rs:48]: Actually, both id and name are ok but it is more convenient for implementing this trait based on meta client.

**Review by waynexia — APPROVED** [review]
👍

**Pull request merged**